instrumentation/kafka-streams/src/main/java/brave/kafka/streams/TracingPredicate.java

```diff
@@ -0,0 +1,43 @@
+package brave.kafka.streams;
+
+import brave.Span;
+import brave.Tracer;
+import org.apache.kafka.streams.KeyValue;
+import org.apache.kafka.streams.kstream.Predicate;
+
+class TracingPredicate<K, V> extends AbstractTracingTransformer<K, V, KeyValue<K, V>> {
+
+  final KafkaStreamsTracing kafkaStreamsTracing;
+  final String spanName;
+  final Predicate<K, V> delegatePredicate;
+
+  TracingPredicate(KafkaStreamsTracing tracing, String spanName,
+      Predicate<K, V> delegatePredicate) {
+    this.kafkaStreamsTracing = tracing;
+    this.spanName = spanName;
+    this.delegatePredicate = delegatePredicate;
+  }
+
+  @Override
+  public KeyValue<K, V> transform(K key, V value) {
+    Span span = kafkaStreamsTracing.tracing.tracer().nextSpan();
+    if (!span.isNoop()) {
+      span.name(spanName);
+      span.start();
+    }
+
+    try (Tracer.SpanInScope ws = kafkaStreamsTracing.tracing.tracer().withSpanInScope(span)) {
+      if (delegatePredicate.test(key, value)) {
+        return KeyValue.pair(key, value);
+      } else {
+        span.annotate(KafkaStreamsTags.KAFKA_STREAMS_FILTERED_TAG);
```

Comment:
We should be careful about `filter` semantics. I would say a tag with filtered=true/false will be better aligned with KStreams#filter semantics: https://github.com/apache/kafka/blob/trunk/streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java#L59

Also, I'd recommend to have a filterNot variable, similar to https://github.com/apache/kafka/blob/trunk/streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamFilter.java#L27 to support `filterNot` as well.

Reply:
> We should be careful about `filter` semantics. I would say a tag with filtered=true/false will be better aligned with KStreams#filter semantics: https://github.com/apache/kafka/blob/trunk/streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java#L59

If we go for `filtered=true|false` that means that every span running through this filter will be tagged. Should we be concerned with the amount of data this generates ? Previously only the records that get filtered out would get annotated, seemed more logical to me.

Reply:
That will depends on the nature of the stream and messages, in some scenarios number of messages with `filtered=false` > `filtered=true`, others `filtered=true` > `filtered=false`.

Based on this, tagging every span is ok IMO.

Reply:
:+1: